qiime2/core/cache.py

```diff
@@ -859,19 +887,45 @@ def load(self, key):
         >>> test_dir.cleanup()
         """
         with self.lock:
-            try:
-                key_values = self.read_key(key)
-                path = self.data / key_values['data']
-            except TypeError as e:
+            key_values = self.read_key(key)
+
+            if 'order' in key_values:
+                return self.load_collection(key)
```

Comment:
This must be for the CLI's benefit? I would generally expect an error, since the return type is dramatically different, but I see why it would be significantly simpler to just get _something_ out of the cache if it's real.

Reply:
Yes, this is for the CLI's benefit. I didn't really want it to work this way, but it makes the CLI so much simpler.

Reply:
Follow up on this, should save dispatch to save_collection or error? It would be nice to keep this behavior consistent. I can also always make it so load doesn't dispatch to load_collection, but IIRC that was looking like it was going to add a bit on annoyance on the CLI, but should be completely doable.

Reply:
I'm honestly inclined to just kinda have save/load defer to save/load collection. It is nice to make people be explicit but idk it just feels like quality of life in this instance